lib/streamlit/runtime/metrics_util.py

```diff
@@ -222,30 +228,35 @@ def wrap(*args, **kwargs):
             and len(ctx.tracked_commands)
             < _MAX_TRACKED_COMMANDS  # Prevent too much memory usage
         )
+        command_telemetry: Union[Command, None] = None
 
-        # Deactivate tracking to prevent calls inside already tracked commands
-        if ctx:
+        if ctx and tracking_activated:
+            # Deactivate tracking to prevent calls inside already tracked commands
             ctx.command_tracking_deactivated = True
+            try:
+                command_telemetry = _get_command_telemetry(callable, *args, **kwargs)
+
+                if (
+                    command_telemetry.name not in ctx.tracked_commands_counter
+                    or ctx.tracked_commands_counter[command_telemetry.name]
+                    < _MAX_TRACKED_PER_COMMAND
+                ):
+                    ctx.tracked_commands.append(command_telemetry)
+                ctx.tracked_commands_counter.update([command_telemetry.name])
+            except Exception as ex:
+                # Always capture all exceptions since we want to make sure that
+                # the telemetry never causes any issues.
+                LOGGER.debug("Failed to collect command telemetry", exc_info=ex)
 
-        exec_start = timer()
         result = callable(*args, **kwargs)
 
         # Activate tracking again
         if ctx:
```

Comment:
Should this fragment be in a finally block? Thanks to this, when an error in the line: `result = callable(*args, **kwargs)` occurs, the original state will be properly restored.

Reply:
What if the exception is caught in the user code?

Reply:
Oh yes indeed, I think this is a valid scenario 👍

Reply:
Put it into `finally` 👍